controller/kubernetes_controller.go

```diff
@@ -387,3 +432,64 @@ func (kc *KubernetesController) detectWorkload(p *v1.Pod, ks *types.KubernetesSt
 	}
 	return "", ""
 }
+
+func (kc *KubernetesController) cleanupVolumeAttachment(volume *longhorn.Volume, ks *types.KubernetesStatus) {
+	// PV and PVC should exist. Pod should be Pending
+	if !(ks.PVStatus == string(v1.VolumeBound) && ks.PVCName != "" && ks.LastPVCRefAt == "") {
+		return
+	}
+	if !(ks.PodName != "" && ks.PodStatus == string(v1.PodPending) && ks.LastPodRefAt == "") {
+		return
+	}
+
+	va, err := kc.getVolumeAttachment(ks)
+	if err != nil {
+		logrus.Errorf("failed to get VolumeAttachment in cleanupVolumeAttachment: %v", err)
+		return
+	}
+	if va == nil {
+		return
+	}
+
+	cleanupFlag := false
+	n, err := kc.nLister.Get(va.Spec.NodeName)
```

Comment:
Can you check if we can use `datastore.IsNodeDownOrDeleted()` here? Then you don't need to node informer.